test/k8sutil/fake.go

```diff
@@ -105,6 +110,31 @@ type DiscoveryClient struct {
 	APIGroup  []*metav1.APIGroup
 }
 
+// OpenAPISchema implements discovery.DiscoveryInterface.
+func (d *DiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
+	panic("unimplemented")
+}
+
+// OpenAPIV3 implements discovery.DiscoveryInterface.
+func (d *DiscoveryClient) OpenAPIV3() openapi.Client {
+	panic("unimplemented")
+}
+
+// RESTClient implements discovery.DiscoveryInterface.
+func (d *DiscoveryClient) RESTClient() rest.Interface {
+	panic("unimplemented")
+}
+
+// ServerVersion implements discovery.DiscoveryInterface.
+func (d *DiscoveryClient) ServerVersion() (*version.Info, error) {
+	panic("unimplemented")
+}
+
+// WithLegacy implements discovery.DiscoveryInterface.
+func (d *DiscoveryClient) WithLegacy() discovery.DiscoveryInterface {
```

Comment:
These functions were missing from our mock. The discovery package defines a ton of different interfaces. I changed from a really scenario-specific one to a more general one.